Performing a no-op animated resize cancels any pending resize immediately

hortont424 opened the pull request:
#### b18af70697f4da911ed70867a325248bd9949367
<pre>
Performing a no-op animated resize cancels any pending resize immediately
<a href="https://bugs.webkit.org/show_bug.cgi?id=242233">https://bugs.webkit.org/show_bug.cgi?id=242233</a>

Reviewed by Wenson Hsieh.

* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _beginAnimatedResizeWithUpdates:]):
_beginAnimatedResizeWithUpdates cancels the resize and completes immediately
if the update block didn&apos;t change any relevant state. For clients who want
to do multiple animated resizes in a row, this can result in a discontinuity
if a impactful animated resize is performed, followed by a no-op animated
resize before the results of the first resize arrive: this results in
the first resize being cancelled, and the web view being shown in an
incompletely-updated state. To avoid this, just avoid cancelling no-op
resizes when there is already a pending resize.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/AnimatedResize.mm:
(TEST):
Add a test. Also move all of the other tests into an AnimatedResize test group
to make it easy to run them together. Add a missing `_endAnimatedResize` to
ResizeWithContentHiddenWithSubsequentNoOpResizeCompletes, because previously
it was depending on this bug.

Canonical link: <a href="https://commits.webkit.org/252106@main">https://commits.webkit.org/252106@main</a>
</pre>

grorg commented on Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
```diff
@@ -826,6 +826,9 @@ static void changeContentOffsetBoundedInValidRange(UIScrollView *scrollView, Web
 {
     auto updateID = layerTreeTransaction.dynamicViewportSizeUpdateID();
+
+    WTFLogAlways("THORTON %f: _didCommitLayerTreeDuringAnimatedResize update %llu want %llu", MonotonicTime::now().secondsSinceEpoch().value(), updateID.value_or(0), _currentDynamicViewportSizeUpdateID);
```
Not sure you want THORTON in here.

grorg replied:
Or was that just to debug the timeout on EWS?

whsieh replied:
(From the commit message, this is temporary logging to try and suss out the timeout on EWS)

hortont424 replied:
I do!!

hortont424 replied:
(two commits, one is the already-reviewed by wenson patch, one is "why in the world is it failing constantly on EWS")

hortont424 committed:
Performing a no-op animated resize cancels any pending resize immediately
https://bugs.webkit.org/show_bug.cgi?id=242233

Reviewed by Wenson Hsieh.

* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(-[WKWebView _beginAnimatedResizeWithUpdates:]):
_beginAnimatedResizeWithUpdates cancels the resize and completes immediately
if the update block didn't change any relevant state. For clients who want
to do multiple animated resizes in a row, this can result in a discontinuity
if a impactful animated resize is performed, followed by a no-op animated
resize before the results of the first resize arrive: this results in
the first resize being cancelled, and the web view being shown in an
incompletely-updated state. To avoid this, just avoid cancelling no-op
resizes when there is already a pending resize.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/AnimatedResize.mm:
(TEST):
Add a test. Also move all of the other tests into an AnimatedResize test group
to make it easy to run them together. Add a missing `_endAnimatedResize` to
ResizeWithContentHiddenWithSubsequentNoOpResizeCompletes, because previously
it was depending on this bug.

Canonical link: https://commits.webkit.org/252106@main
